channels/channel.nos/nos2010/chn_nos2010.py

```diff
@@ -1331,6 +1332,81 @@ def update_video_item_live(self, item):
         item.complete = True
         return item
 
+    def create_iptv_streams(self, parameter_parser):
+        """ Fetch the available live channels using EPG endpoint and format them into JSON-STREAMS
+
+        :param ActionParser parameter_parser: a ActionParser object to is used to parse and
+                                                   create urls
+
+        :return: Formatted stations
+        :rtype: list
+        """
+
+        epg_data = JsonHelper.loads(UriHandler.open(datetime.datetime.now().strftime("https://start-api.npo.nl/epg/%Y-%m-%d?type=tv"),
+            no_cache=True,
+            additional_headers=self.__jsonApiKeyHeader))
+
+        parent_item = MediaItem("Live", "https://www.npostart.nl/live", media_type=mediatype.FOLDER)
+        items = []
+        iptv_streams = []
+
+        for stations in epg_data["epg"]:
+            livestream = stations["channel"]["liveStream"]
+            item = MediaItem(livestream["title"], livestream["shareUrl"], media_type=mediatype.VIDEO)
+            items.append(item)
+
+            iptv_streams.append(dict(
+                id=livestream["id"],
+                name=livestream["title"],
+                logo=livestream["images"]["original"]["formats"]["tv"]["source"],
+                group=self.channelName,
+                stream=parameter_parser.create_action_url(self, action=action.PLAY_VIDEO, item=item, store_id=parent_item.guid),
+            ))
+        
+        parameter_parser.pickler.store_media_items(parent_item.guid, parent_item, items)
+
+        return iptv_streams
+
+    def create_iptv_epg(self, parameter_parser):
+        """ Fetch the EPG using the EPG endpoint and format it into JSON-EPG
+
+        :param ActionParser parameter_parser: a ActionParser object to is used to parse and
+                                                   create urls
+
+        :return: Formatted stations
+        :rtype: dict
+        """
+
+        parent = MediaItem("EPG", "https://start-api.npo.nl/epg/", media_type=mediatype.FOLDER)
+        iptv_epg = dict()
+        media_items = []
+
+        start = datetime.datetime.now() - datetime.timedelta(days=3)
+        for i in range(0, 7, 1):
+            air_date = start + datetime.timedelta(i)
+            data = UriHandler.open(air_date.strftime("https://start-api.npo.nl/epg/%Y-%m-%d?type=tv"),
+                no_cache=True,
+                additional_headers=self.__jsonApiKeyHeader)
+
+            json_data = JsonHelper.loads(data)
+            for epg_item in json_data["epg"]:
+                id = epg_item["channel"]['liveStream']['id']
+                iptv_epg[id]=iptv_epg.get(id, [])
+                for program in epg_item["schedule"]:
+                    media_item = MediaItem(program["program"]["title"], program["program"]["id"], media_type=mediatype.EPISODE)
```

Comment:
Again here, I think you need to add `isLive` at least.

Reply:
These are the Video-on-demand streams, so they are not live.
However, I will check if I can adopt the existing information concerning DRM and Geoblock.

Reply:
Makes sense that it is not live.